src/dbnode/server/server.go

```diff
@@ -135,6 +136,9 @@ type RunOptions struct {
 	// interrupt and shutdown the server.
 	InterruptCh <-chan error
 
+	// QueryStatsTracker exposes an interface for tracking query stats.
+	QueryStatsTracker *stats.QueryStatsTracker
```

Comment:
`QueryStatsTracker` is already an interface (i.e. ptr type) so don't need to make this pointer as well. 

Even without the ptr you can check if `runOpts.QueryStatsTracker != nil`.

So just make this  `QueryStatsTracker stats.QueryStatsTracker`